pkg/doc/verifiable/credential_ldp_test.go

```diff
@@ -208,6 +211,191 @@ func TestNewCredentialFromLinkedDataProof_JSONLD_Validation(t *testing.T) {
+			return &sigverifier.PublicKey{
+				Type: "JwsVerificationKey2020",
+				JWK: &jose.JWK{
+					JSONWebKey: gojose.JSONWebKey{
+						Algorithm: "EdDSA",
+						Key:       rv,
+					},
+					Crv: "Ed25519",
+					Kty: "OKP",
+				},
+			}, nil
+		}),
+		WithExtraJSONLDContexts("https://trustbloc.github.io/context/vc/credentials-v1.jsonld"),
+		WithStrictValidation())
+
+	require.NoError(t, err)
+	require.NotNil(t, vcWithLdp)
+}
+
+func TestExtraContextWithLDP(t *testing.T) {
+	r := require.New(t)
+
+	vcJSON := `
+{
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://trustbloc.github.io/context/vc/examples-v1.jsonld"
+  ],
+  "id": "http://example.edu/credentials/3732",
+  "type": ["VerifiableCredential", "SupportingActivity"],
+  "issuer": "https://example.edu/issuers/14",
+  "issuanceDate": "2010-01-01T19:23:24Z",
+  "credentialSubject": {
+    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21"
+  },
+  "credentialStatus": {
+    "id": "https://example.edu/status/24",
+    "type": "CredentialStatusList2017"
+  }
+}`
+
+	pubKey, privKey, err := ed25519.GenerateKey(rand.Reader)
+	r.NoError(err)
+
+	sigSuite := ed25519signature2018.New(
+		suite.WithSigner(getEd25519TestSigner(privKey)),
+		suite.WithVerifier(ed25519signature2018.NewPublicKeyVerifier()))
+
+	ldpContext := &LinkedDataProofContext{
+		SignatureType:           "Ed25519Signature2018",
+		SignatureRepresentation: SignatureProofValue,
+		Suite:                   sigSuite,
+		VerificationMethod:      "did:example:123456#key1",
+	}
+
+	vc, _, err := NewCredential([]byte(vcJSON))
+	r.NoError(err)
+
+	err = vc.AddLinkedDataProof(ldpContext)
+	r.NoError(err)
+
+	vcBytes, err := json.Marshal(vc)
+	r.NoError(err)
+
+	vcWithLdp, _, err := NewCredential(vcBytes,
+		WithEmbeddedSignatureSuites(sigSuite),
+		WithPublicKeyFetcher(SingleKey(pubKey, kms.ED25519)),
+		WithStrictValidation())
+	r.NoError(err)
+	r.Equal(vc, vcWithLdp)
+	r.NotNil(vcWithLdp)
+
+	// Drop https://trustbloc.github.io/context/vc/examples-v1.jsonld context where
+	// SupportingActivity and CredentialStatusList2017 are defined.
+	vcMap, err := toMap(vcBytes)
+	r.NoError(err)
+
+	vcMap["@context"] = baseContext
+	vcBytes, err = json.Marshal(vcMap)
+	r.NoError(err)
+
+	vcWithLdp, _, err = NewCredential(vcBytes,
+		WithEmbeddedSignatureSuites(sigSuite),
+		WithPublicKeyFetcher(SingleKey(pubKey, kms.ED25519)),
+		WithStrictValidation())
+	r.Error(err)
+	r.EqualError(err, "decode new credential: check embedded proof: check linked data proof: ed25519: invalid signature")
+	r.Nil(vcWithLdp)
+
+	// Use extra context.
+	vcWithLdp, _, err = NewCredential(vcBytes,
+		WithEmbeddedSignatureSuites(sigSuite),
+		WithPublicKeyFetcher(SingleKey(pubKey, kms.ED25519)),
+		WithExtraJSONLDContexts("https://trustbloc.github.io/context/vc/examples-v1.jsonld"),
+		WithStrictValidation())
+	r.NoError(err)
+	r.NotNil(vcWithLdp)
+
+	// Use extra context.
+	vcWithLdp, _, err = NewCredential(vcBytes,
```

Comment:
An example of using extra remote context https://trustbloc.github.io/context/vc/examples-v1.jsonld to use custom fields like credential status type `CredentialStatusList2017`  to be used in Linked Data Signature verification data and also pass strict VC validation.